realm/src/main/java/io/realm/internal/Context.java

```diff
@@ -52,6 +57,17 @@ public void executeDelayedDisposal() {
                 TableQuery.nativeClose(nativePointer);
             }
             abandonedQueries.clear();
+
+            cleanRows();
+        }
+    }
+
+    public void cleanRows() {
+        NativeObjectReference reference = (NativeObjectReference) referenceQueue.poll();
+        while (reference != null) {
+            UncheckedRow.nativeClose(reference.nativePointer);
+            rowReferences.remove(reference);
+            reference = (NativeObjectReference) referenceQueue.poll();
```

Comment:
if `Context.isFinalized` is usefull, we probably need to check it in this function as well.